### PR TITLE
Update aws cli from v1 to v2

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#126956](https://progress.opensuse.org/issues/126956)
- Verification run: [create_hdd_autoyast_pc](https://pdostal-server.suse.cz/tests/4714), [img_proof](https://pdostal-server.suse.cz/tests/4721)

### CODE REVIEW

**Review by ricardobranco777 — APPROVED** [review]
The GPG signature check is a bit overkill, IMO.  Anyway LGTM.

File: tests/publiccloud/prepare_tools.pm [hunk 1]
```diff
@@ -71,14 +71,23 @@ sub run {
...
     # Install AWS cli
-    install_in_venv('aws', requirements => 1);
+    my $aws_version = '2.12.3';
+    # Download and import the AWS public PGP key
+    assert_script_run(sprintf('curl -f -v %s/data/publiccloud/aws.asc -o /tmp/aws.asc', autoinst_url()));
+    assert_script_run('gpg --import /tmp/aws.asc');
+    # Download the aws cli binary, its signature and verify those
+    assert_script_run("curl https://awscli.amazonaws.com/awscli-exe-linux-x86_64-$aws_version.zip -o /tmp/awscliv2.zip");
+    assert_script_run("curl https://awscli.amazonaws.com/awscli-exe-linux-x86_64-$aws_version.zip.sig -o /tmp/awscliv2.sig");
+    assert_script_run('gpg --verify /tmp/awscliv2.sig /tmp/awscliv2.zip');
```
Comment (grisu48) [inline tests/publiccloud/prepare_tools.pm:88]:
```suggestion
    assert_script_run('gpg --verify /tmp/awscliv2.sig /tmp/awscliv2.zip', fail_message => "gpg check of downloaded installation file failed");
```

Just to nitpick :wink:

File: tests/publiccloud/prepare_tools.pm [hunk 2]
```diff
@@ -71,14 +71,23 @@ sub run {
...
     assert_script_run('docker ps');
 
     # Install AWS cli
-    install_in_venv('aws', requirements => 1);
+    my $aws_version = '2.12.3';
+    # Download and import the AWS public PGP key
+    assert_script_run(sprintf('curl -f -v %s/data/publiccloud/aws.asc -o /tmp/aws.asc', autoinst_url()));
+    assert_script_run('gpg --import /tmp/aws.asc');
+    # Download the aws cli binary, its signature and verify those
+    assert_script_run("curl https://awscli.amazonaws.com/awscli-exe-linux-x86_64-$aws_version.zip -o /tmp/awscliv2.zip");
```
Comment (grisu48) [inline tests/publiccloud/prepare_tools.pm:86]:
```suggestion
    script_retry("curl https://awscli.amazonaws.com/awscli-exe-linux-x86_64-$aws_version.zip -o /tmp/awscliv2.zip", retry => 3, delay => 60);
```

As a suggestion

File: tests/publiccloud/prepare_tools.pm [hunk 3]
```diff
@@ -71,14 +71,23 @@ sub run {
...
+    assert_script_run("curl https://awscli.amazonaws.com/awscli-exe-linux-x86_64-$aws_version.zip.sig -o /tmp/awscliv2.sig");
```
Comment (grisu48) [inline tests/publiccloud/prepare_tools.pm:87]: Same as above, I would retry each `curl` call for increased stability.

**Review by grisu48 — APPROVED** [review]
Nice addition! I left some suggestion but it's up to you to pick them or not. Just nitpicking.

File: tests/publiccloud/prepare_tools.pm [hunk 4]
```diff
@@ -71,14 +71,23 @@ sub run {
...
     record_info('python', script_output('python --version'));
     systemctl('enable --now docker');
     assert_script_run('podman ps');
     assert_script_run('docker ps');
 
     # Install AWS cli
-    install_in_venv('aws', requirements => 1);
+    my $aws_version = '2.12.3';
+    # Download and import the AWS public PGP key
+    assert_script_run(sprintf('curl -f -v %s/data/publiccloud/aws.asc -o /tmp/aws.asc', autoinst_url()));
```
Comment (asmorodskyi) [inline tests/publiccloud/prepare_tools.pm:83]: why we importing this key from our repo ? usually such things are shipped together with package ? from where you got this key ?

Comment (pdostal) [reply tests/publiccloud/prepare_tools.pm:83]: From their [official documentation](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).
image

I decided to download it as it's more 'long term' compared to the package itself and also it plays role in the verification process of the package so I thought that it's better to not download everything.

**Comment by pdostal** [comment]
@asmorodskyi there were suggestions from Felix.

**Pull request merged**